plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/GoogleApiFactory.java

```diff
@@ -0,0 +1,185 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.googleapis.internal;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.ConnectionFactory;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.api.services.appengine.v1.Appengine;
+import com.google.api.services.appengine.v1.Appengine.Apps;
+import com.google.api.services.cloudresourcemanager.CloudResourceManager;
+import com.google.api.services.cloudresourcemanager.CloudResourceManager.Projects;
+import com.google.cloud.tools.eclipse.googleapis.IGoogleApiFactory;
+import com.google.cloud.tools.eclipse.util.CloudToolsInfo;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.Proxy.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.eclipse.core.net.proxy.IProxyChangeEvent;
+import org.eclipse.core.net.proxy.IProxyChangeListener;
+import org.eclipse.core.net.proxy.IProxyData;
+import org.eclipse.core.net.proxy.IProxyService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * Class to obtain various Google cloud Platform related APIs.
+ */
+@Component
+public class GoogleApiFactory implements IGoogleApiFactory {
+
+  private static final int DEFAULT_TIMEOUT_MS = 1000;
+
+  private static final Logger logger = Logger.getLogger(GoogleApiFactory.class.getName());
+
+  private JsonFactory jsonFactory;
+  private Map<String, HttpTransport> transports = new HashMap<>();
+
+  private IProxyService proxyService;
+  private GoogleApiUrls googleApiUrls;
+
+  public GoogleApiFactory() {
+    this(new GoogleApiUrls());
+  }
+
+  @VisibleForTesting
+  GoogleApiFactory(GoogleApiUrls googleApiUrls) {
+    this.googleApiUrls = googleApiUrls;
+  }
+
+  @Activate
+  public void init() {
+    Preconditions.checkNotNull(proxyService, "proxyService is null");
+    jsonFactory = new JacksonFactory();
+    buildTransports();
+    proxyService.addProxyChangeListener(new IProxyChangeListener() {
```

Comment:
I think this would be better if we made this service optionally dependent on the IProxyService (i.e., `ReferenceCardinality.OPTIONAL`: 0..1).  If there is no proxy service, we just treat everything like DIRECT.  (I believe the IProxyService is actually registered programmatically in `org.eclipse.core.net`'s activator and so if that hasn't been activated — which is unlikely, but possible — then our service won't be available.)